innovation_sweet_spots/analysis/lda_modelling_utils.py

```diff
@@ -337,6 +335,90 @@ def make_pyLDAvis(topic_model: LDAModel, output_filepath: PathLike):
     )
 
 
+def topic_model_param_search(
+    n_topics: list,
+    max_iterations: int,
+    n_iters_logging: int,
+    corpus: tp.utils.Corpus,
+    random_seeds: list,
+) -> pd.DataFrame:
+    """Train a tomotopy LDAModel with different numbers of topics.
+    After a specfiied number of iterations, record the coherence scores.
+
+    Args:
+        n_topics: List of number of topics to search through
+        max_iterations: Max number of model training iterations
+        n_iters_logging: Log results after this number of iterations
+        corpus: Text corpus
+        random_seeds: Param search will be performed for each
+            random seed specified and then the results averaged
+
+    Returns:
+        Dataframe with columns for iterations, n_topics, coherence_score.
+    """
+    iteration_record = []
+    n_topics_record = []
+    coherence_score_record = []
+    random_seed_record = []
+    for random_seed in tqdm(random_seeds):
+        for k in tqdm(n_topics):
+            model = tp.LDAModel(k=k, corpus=corpus, min_df=1, seed=random_seed)
+            for i in tqdm(
+                range(0, max_iterations + n_iters_logging, n_iters_logging),
```

Comment:
```suggestion
                range(0, max_iterations, n_iters_logging),
```

I think if you have `max_iterations + n_iters_logging` it will run the training for one cycle longer (cycle = `n_iters_logging`). When you plot the results, it starts with "0" iterations, but that's actually already `n_iters_logging` number of iterations.